// Copyright (c) 2019 Tesla, Inc.
//
// Licensed under the Apache License, Version 2.0 (the License);
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an AS IS BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Author: Michael Tesla (michaeltesla1995@gmail.com)
// Date: Tue Oct 29 23:15:39 CST 2019

syntax = "proto3";

package tesla.authentication;

import "authentication/general_response.proto";

option cc_generic_services = true;

message RegisterUserRequest {
  string user_name = 1; 
  string password = 2;
  string email = 3;
  string mobile_phone = 4;
};

service RegisterUserService {
  rpc RegisterUser (RegisterUserRequest) returns (GeneralResponse);
};
